lib/jarvis/mixins/pool_delegate.rb

```diff
@@ -0,0 +1,27 @@
+require "jarvis/workpool"
+
+module Jarvis module Mixins module PoolDelegate
+  def pool_delegate(pool_name, &callback)
+    lambda do |request|
+      begin
+        pool = WorkPool.fetch(pool_name)
+        pool.post do
+          # TODO(sissel): Note what command is being executed and at what time.
+          # TODO(sissel): Redirect $stdout and $stderr
+          begin
+            callback.call(request)
+          rescue => e
+            # TODO(sissel): Mark this job as failed
+            request.reply(t("unhandled exception", :exception => e))
+            request.reply(e.backtrace.join("\n"))
```

Comment:
I think lita chose the lowest denominator and goes with plain text by default.
I will investigate to see how hard is to change the adapter to support it OR create our own reply that use the api instead of the xmpp.